network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/detail_page.py

```diff
@@ -115,64 +107,144 @@ class RCCDetailPage(BasePage):
+                    "author_profile",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_content_types",
+            [
+                index.RelatedFields(
+                    "content_type",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_curricular_areas",
+            [
+                index.RelatedFields(
+                    "curricular_area",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+        index.RelatedFields(
+            "related_topics",
+            [
+                index.RelatedFields(
+                    "rcc_topic",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
     ]
 
-    # def get_context(self, request):
-    #     context = super().get_context(request)
-    #     context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
-    #     context["rcc_authors"] = self.get_rcc_authors()
-    #     context["rcc_author_names"] = self.get_rcc_author_names()
-    #     context["content_type_names"] = self.get_related_content_types_names()
-    #     return context
+    def get_context(self, request):
+        context = super().get_context(request)
+        context["authors_index"] = authors_index.RCCAuthorsIndexPage.objects.first()
+        context["rcc_authors"] = self.get_rcc_authors()
+        context["rcc_author_names"] = self.get_rcc_author_names()
+        context["content_type_names"] = self.get_related_content_types_names()
+        return context
 
-    # def get_rcc_authors(self):
-    #     rcc_author_profiles = localize_queryset(
-    #         Profile.objects.prefetch_related("authored_rcc_entries").filter(authored_rcc_entries__rcc_detail_page=self)
-    #     )
-    #     return rcc_author_profiles
+    def get_rcc_authors(self):
+        rcc_author_profiles = localize_queryset(
+            Profile.objects.prefetch_related("authored_rcc_articles").filter(
+                authored_rcc_articles__rcc_detail_page=self
+            )
+        )
+        return rcc_author_profiles
 
-    # def get_rcc_author_names(self):
-    #     return [ra.author_profile.name for ra in self.rcc_authors.all()]
+    def get_rcc_author_names(self):
+        return [ra.author_profile.name for ra in self.rcc_authors.all()]
 
-    # def get_related_content_types_names(self):
-    #     return [ct.content_type.name for ct in self.related_content_types.all()]
+    def get_related_content_types_names(self):
+        return [ct.content_type.name for ct in self.related_content_types.all()]
 
     def get_banner(self):
         return self.get_parent().specific.get_banner()
+
+
+class RCCDetailLink(wagtail_models.TranslatableMixin, wagtail_models.Orderable):
+    rcc_detail_page = cluster_fields.ParentalKey(
```

Comment:
Do we need the prefix here?

Reply:
We don't, but the research hub one had, so I was trying to keep it consistent

Reply:
I understand. Consistency is good, but I think we should probably rather clean up research hub too. There are many more taxonomies and relations now and there is not really anything that makes an `rcc_link` special from other links, etc.

Reply:
#10697